packages/blog-starter-kit/themes/enterprise/pages/[slug].tsx

```diff
@@ -192,63 +187,45 @@ export const getStaticProps: GetStaticProps<Props, Params> = async ({ params })
 	if (!params) {
 		throw new Error('No params');
 	}
-	const data = await request<SinglePostByPublicationQuery, SinglePostByPublicationQueryVariables>(
-		process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT,
-		SinglePostByPublicationDocument,
-		{
-			host: process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST,
-			slug: params.slug,
-		},
-	);
 
-	// Extract the post data from the GraphQL response
-	const publication = data.publication;
-	if (!publication) {
+	const endpoint = process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT;
+	const host = process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST;
+	const slug = params.slug;
+
+	const postData = await request(endpoint, SinglePostByPublicationDocument, { host, slug });
+
+	if (postData.publication?.post) {
 		return {
-			notFound: true,
+			props: {
+				type: 'post',
+				post: postData.publication.post,
+				publication: postData.publication,
+			},
 			revalidate: 1,
 		};
 	}
-	const post = publication.post;
-	if (!post) {
-		const staticPageData = await request<PageByPublicationQuery, PageByPublicationQueryVariables>(
-			process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT,
-			PageByPublicationDocument,
-			{
-				host: process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST,
-				slug: params.slug,
-			},
-		);
 
-		const page = staticPageData.publication?.staticPage;
-		if (!page) {
-			return {
-				notFound: true,
-				revalidate: 1,
-			};
-		}
+	const pageData = await request(endpoint, PageByPublicationDocument, { host, slug });
+
+	if (pageData.publication?.staticPage) {
 		return {
 			props: {
-				post: null,
-				page,
-				publication,
+				type: 'page',
+				page: pageData.publication.staticPage,
+				publication: pageData.publication,
 			},
 			revalidate: 1,
 		};
 	}
 
 	return {
-		props: {
-			post,
-			page: null,
-			publication,
-		},
+		notFound: true,
 		revalidate: 1,
 	};
 };
 
 export const getStaticPaths: GetStaticPaths = async () => {
-	const data = await request<SlugPostsByPublicationQuery, SlugPostsByPublicationQueryVariables>(
```

Comment:
Could you please provide insight into the rationale behind the removal of the Typescript generic checks in this instance?

Reply:
`Query` and `Variables` types are completely inferred from `SlugPostsByPublicationDocument`, they don't need to be explicitly used in generics.

<img width="597" alt="image" src="https://github.com/Hashnode/starter-kit/assets/1881266/a08de3c0-4151-4611-aa2c-49fa0c6e754c">

Reply:
Got it. Thanks for the explanation @iamandrewluca .